src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -455,6 +455,7 @@ internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestCon
                 {
                     if (RefreshNeeded(scriptFile.ClientUri)) {
                         await connectionService.TryRefreshAuthToken(scriptFile.ClientUri);
+                        await requestContext.SendResult(null);
```

Comment:
why sending null?

Reply:
I want to use offline intellisense if the auth token needs to be refreshed

Reply:
what is the expected behavior here? I thought we want to wait for the refresh to complete and then have the correct intellisense send back to the UI.

Reply:
@alanrenmsft We can't do that because of the synchronous processing issue that you have the PR out to start addressing. My understanding is that this is a temporary quick fix to unblock this until we can fully fix the synchronous processing issue - at which point this stuff should be removed and replaced with a Request call

Reply:
The disabling intellisense is part of the issue in the ICM, they're saying that the log files were growing very quickly once a token expired because intellisense kept trying to autofill but could not due to the expired token

The refresh token is able to be refreshed in the background, based on an access token that expires every couple months, so no user interaction should be necessary.

Reply:
Refresh tokens aren't guaranteed to be valid - they can be revoked or time out for example. (I think there's also an AAD thing you can change to require user interaction each time - it would be good to verify this)

https://docs.microsoft.com/en-us/azure/active-directory/develop/refresh-tokens

So I think we should at least be considering the possibility of this happening. I would suggest looking at how other implementations in VS Code handles this.

Reply:
Ping - did you look into this at all? It's just going to keep trying to refresh constantly right? I'm not sure that's the behavior we want - it feels like we should cut it off at some point and just stop trying

Reply:
I have added a dictionary to keep track of the state - if the token is in the "Updating" state, I won't request another token